src/utils/narrow.js

```diff
@@ -123,17 +124,19 @@ export const pmNarrowFromUsersUnsafe = (recipients: $ReadOnlyArray<UserOrBot>):
 export const pm1to1NarrowFromUser = (user: UserOrBot): Narrow =>
   pmNarrowInternal(pmKeyRecipientsFor1to1(user.user_id));
 
-export const specialNarrow = (operand: string): Narrow => {
-  if (operand === 'starred') {
-    return Object.freeze({ type: 'starred' });
-  }
-  if (operand === 'mentioned') {
-    return Object.freeze({ type: 'mentioned' });
-  }
-  if (operand === 'private') {
-    return Object.freeze({ type: 'all-pm' });
+export const specialNarrow = (operand: 'starred' | 'mentioned' | 'dm' | 'private'): Narrow => {
+  switch (operand) {
+    case 'starred':
+      return Object.freeze({ type: 'starred' });
+    case 'mentioned':
+      return Object.freeze({ type: 'mentioned' });
+    case 'dm': // new in server-future; means the same as 'private'
+    case 'private':
+      return Object.freeze({ type: 'all-pm' });
```

Comment:
Let's treat the argument to `specialNarrow` as part of our own internal API, and not have it be made more complicated by variation between servers (or eras of messages.) So it can take just `dm`, and let it be the responsibility of the parsing code in `internalLinks.js` to interpret both `/is/private` and `/is/dm` that way.